### PR TITLE
Create customer schema

### PR DESCRIPTION
**Pull request by mwj10**
Pull request for Creating Customer Schema

### CODE REVIEW

File: tests/test_models.py [hunk 1]
```diff
@@ -1,37 +1,252 @@
...
+        data = {"id": 1, "first_name": "Kitty", "email": "kitty@gmail.com"}
+        customer = Customer()
+        self.assertRaises(DataValidationError, customer.deserialize, data)
+    
+    def test_deserialize_bad_data(self):
+        """It should not deserialize bad data"""
+        data = "this is not a dictionary"
+        customer = Customer()
+        self.assertRaises(DataValidationError, customer.deserialize, data)
+
```
Comment (Akshama) [inline tests/test_models.py:176]: Can add tests for deserializing data with key error (empty JSON or JSON without an id) and data which is not JSON (type error).

File: tests/test_models.py [hunk 2]
```diff
@@ -1,37 +1,252 @@
...
+            customer.create()
+        last_name = customers[0].last_name
+        count = len([customer for customer in customers if customer.last_name == last_name])
+        found = Customer.find_by_last_name(last_name)
+        self.assertEqual(found.count(), count)
+        for customer in found:
+            self.assertEqual(customer.last_name, last_name)
+
+    def test_find_by_email(self):
+        """It should Find Customer by Last Name"""
```
Comment (Akshama) [inline tests/test_models.py:219]: Docstring should say email instead of Last Name

**Review by Akshama — CHANGES_REQUESTED** [review]
Great work! Tested out the Database schema and model, works fine. :) 
I've mentioned some review comments for small changes. We can merge once these are committed.
Additionally, could you uncomment lines 28-29 in `docker-compose.yml` as part of this issue?
```
    ports:
       - 5432:5432
```
I had to expose these ports for local testing of the DB.

File: service/models.py [hunk 3]
```diff
@@ -1,89 +1,109 @@
...
     # Table Schema
+    ##############
+
     id = db.Column(db.Integer, primary_key=True)
-    name = db.Column(db.String(63))
+    first_name = db.Column(db.String(255), nullable=False)
+    last_name = db.Column(db.String(255), nullable=False)
+    email = db.Column(db.String(255), nullable=False)
+    password = db.Column(db.String(255), nullable=False)
+    # Phone Number
```
[the remark this answers is not in the document]
Comment (mwj10) [reply service/models.py:39]: Definitely we can do this, but it might be redundant as the date of each log might be already captured when call the logging function. Therefore, we need to check the documentation of the function.

[the remark this answers is not in the document]
Comment (mwj10) [reply service/models.py:39]: I just check the log_handlers.py and I found the following: `formatter = logging.Formatter("[%(asctime)s] [%(levelname)s] [%(module)s] %(message)s", "%Y-%m-%d %H:%M:%S %z")`. Based on this, I think the the time and date is automatically captured by the server along other things.

[the remark this answers is not in the document]
Comment (Akshama) [reply service/models.py:39]: Okay, that's a good point. However, I'm not sure where these logs are being stored. Was checking the gunicorn and Flask documentation, but couldn't find the logs file path. 
I feel it'll be better to include the two fields in Customer for now to handle the REST API calls accordingly since we plan to implement Log Schema in later sprints. Can discuss more in the standup call and resolve this.

File: tests/test_models.py (hunk 1 above)
Comment (Akshama) [reply tests/test_models.py:176]: Covered by the tests already.

File: service/models.py [hunk 4]
```diff
@@ -1,89 +1,109 @@
...
-                "Invalid YourResourceModel: missing " + error.args[0]
-            ) from error
+            raise DataValidationError("Invalid Customer: missing " + error.args[0]) from error
         except TypeError as error:
-            raise DataValidationError(
-                "Invalid YourResourceModel: body of request contained bad or no data - "
-                "Error message: " + error
-            ) from error
+            raise DataValidationError("Invalid pet: body of request contained bad or no data " + str(error)) from error
         return self
```
[the remark this answers is not in the document]
Comment (Akshama) [reply service/models.py:99]: Followed up with Prof.

File: service/models.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (Akshama) [reply service/models.py:39]: Followed up with Prof.
